File: refrence-service/src/main/java/com/kapitonau/ps/refrenceservice/model/ReferenceItemModel.java
package com.kapitonau.ps.refrenceservice.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.util.Map;

@Getter
@Setter
@Entity
@Table(name = "reference_items")
public class ReferenceItemModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "reference_item_id", nullable = false)
    private Long referenceItemId;

    @Column(name = "reference_type", nullable = false, length = Integer.MAX_VALUE)
    private String referenceType;

    @Column(name = "item_code", nullable = false, length = Integer.MAX_VALUE)
    private String itemCode;

    @Column(name = "description", nullable = false)
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, Object> description;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

}

File: reference-service/src/main/kotlin/com/kapitonau/referenceservice/model/ReferenceItemModel.kt
package com.kapitonau.referenceservice.model

import jakarta.persistence.*
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes

@Entity
@Table(name = "reference_item")
open class ReferenceItemModel {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "reference_item_id_gen")
    @SequenceGenerator(
        name = "reference_item_id_gen",
        sequenceName = "reference_item_reference_item_id_seq",
        allocationSize = 1
    )
    @Column(name = "reference_item_id", nullable = false)
    open var referenceItemId: Long? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "reference_type", nullable = false)
    open var referenceType: ReferencesTypeModel? = null

    @Column(name = "item_code", nullable = false, length = Integer.MAX_VALUE)
    open var itemCode: String? = null

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "description", nullable = false)
    open var description: Map<String, String>? = null
}